src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/ServiceResponseParser.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpStatusCodeException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.CaseTransformationException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.InvalidCaseDataException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.model.response.TransformationErrorResponse;
+
+import java.io.IOException;
+
+@Component
+public class ServiceResponseParser {
+
+    protected final ObjectMapper objectMapper;
+
+    protected ServiceResponseParser(
+        ObjectMapper objectMapper
+    ) {
+        this.objectMapper = objectMapper;
+    }
+
+    public void tryParseResponseBodyAndThrow(HttpStatusCodeException exception) throws CaseTransformationException {
+        try {
+            TransformationErrorResponse errorResponse = objectMapper.readValue(
+                exception.getResponseBodyAsByteArray(),
+                TransformationErrorResponse.class
+            );
+
+            throw new InvalidCaseDataException(exception, errorResponse);
+        } catch (IOException ioException) {
+            throw new CaseTransformationException(exception, ioException.getMessage());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [New exception is thrown in catch block, original stack trace may be lost](https://app.codacy.com/app/HMCTS/bulk-scan-orchestrator/pullRequest?prid=4518453)